components/servo/tests/multiprocess.rs

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! Unit tests exercising Servo API functionality with multiprocess mode enabled.
+//!
+//! Since all Servo tests must run serially on the same thread, it is important
+//! that tests never panic. In order to ensure this, use `anyhow::ensure!` instead
+//! of `assert!` for test assertions. `ensure!` will produce a `Result::Err` in
+//! place of panicking.
+
+#[allow(dead_code)]
+mod common;
+
+use std::rc::Rc;
+
+use anyhow::ensure;
+use common::{ServoTest, WebViewDelegateImpl, evaluate_javascript, run_api_tests};
+use servo::{JSValue, WebViewBuilder, run_content_process};
+use servo_config::{opts, prefs};
+
+fn test_multiprocess_preference_observer(servo_test: &ServoTest) -> Result<(), anyhow::Error> {
+    let delegate = Rc::new(WebViewDelegateImpl::default());
+    let webview = WebViewBuilder::new(servo_test.servo())
+        .delegate(delegate.clone())
+        .build();
+
+    let result = evaluate_javascript(servo_test, webview.clone(), "window.gc");
+    ensure!(matches!(result, Ok(JSValue::Undefined)));
+
+    let mut prefs = prefs::get().clone();
+    prefs.dom_servo_helpers_enabled = true;
+    prefs::set(prefs);
+
+    webview.reload();
+
+    let result = evaluate_javascript(servo_test, webview.clone(), "window.gc");
+    ensure!(matches!(result, Ok(JSValue::Object(..))));
```

Comment:
This is permafailing for me in debug:

```
./mach test-unit -d test_multiprocess_preference_observer
```

Reply:
Hmm, I've never been able to reproduce this on macOS.